### DISCUSSION TITLE
keep骑行数据导出为gpx文件

### DISCUSSION THREAD

**Discussion by Similarzhang** [yihong0618/running_page#279] [Q&A] [answered]
您好，是否可能修改keep_sync.py文件使其可以支持导出keep的骑行数据为gpx文件，后续手动上传至garmin connect或strava，非常感谢。

**Comment by yihong0618** [comment, marked as answer]
@Similarzhang 可以的
修改 `keep_sync.py` 中 runninglog 改成 cyclinglog
https://github.com/yihong0618/running_page/blob/166474a76d282012ad93b31ac4443f04e315297b/scripts/keep_sync.py#L21


https://api.gotokeep.com/pd/v3/cyclinglog/

**Reply by Similarzhang** [reply]
感谢回复！我手动替换了 keep_sync.py 中 runninglog 为 cyclinglog，运行代码之后显示如下（替换了手机号和密码），可能是什么问题？

`similar@MacBook-Pro running_page-master % python3 scripts/keep_sync.py 184XXXXXXXX PASSWORD --with-gpx
pares keep ids data since 1970-01-01 00:00:00
0 new keep runs to generate
No tracks found.`

**Comment by yihong0618** [comment]
删掉 data.db 和所有之前生成的 gpx 重试一下。

**Reply by Similarzhang** [reply]
感谢回复。我只使用了keep_sync.py，没有生成gpx。尝试了很多次，每次删掉data.db，还是如上报一样的错误。

**Reply by yihong0618** [reply]
gpx 删了么？我没骑行数据没法测不好意思

**Reply by Similarzhang** [reply]
GPX_OUT内没有生成gpx文件，是指这个吗？谢谢。

**Reply by ben-29** [reply]
修改 keep_sync.py 中的type, 把 running 改成 cycling，再试试？
https://github.com/yihong0618/running_page/blob/a342591c26b410ada3fd509d965cfba3ad22a885/scripts/keep_sync.py#L20
RUN_DATA_API = "https://api.gotokeep.com/pd/v3/stats/detail?dateUnit=all&type=cycling&lastDate={last_date}"

**Reply by Similarzhang** [reply]
成功了！非常感谢，我回头去骑行论坛也宣传一下这个项目。

**Reply by yihong0618** [reply]
还有骑行论坛么。。。

**Reply by Similarzhang** [reply]
哈哈，中文有几个，还有贴吧啥的

**Reply by scelezy** [reply]
你好，我按照你说的 手动替换了 keep_sync.py 中 runninglog 为 cyclinglog，然后更改了type 为cycling，但是出现了这个问题，不知道啥情况
parsing keep id 5633887cc07a219f2caa0e3b_9223370522910643807_cy
Something wrong paring keep id 5633887cc07a219f2caa0e3b_9223370522910643807_cy'NoneType' object is not subscriptable
No tracks found.

**Reply by JiNanPiWang** [reply]
我之前也是和你一样的问题，在测试了RUN_DATA_API返回内容之后，突然就好了，你可以参考一下我keep_sync文件的代码：https://paste.ubuntu.com/p/b3D9sC2bFY/

**Comment by hisenyuan** [comment]
感谢，让我导出了 keep 上的骑行数据

**Reply by yihong0618** [reply]
不客气，骑行的 page 你可以参考 readme 下面推荐的 fork

**Comment by scelezy** [comment]
好的，谢谢Sent from my Galaxy
-------- Original message --------From: JNPW ***@***.***> Date: 26/07/2023  17:47  (GMT+08:00) To: yihong0618/running_page ***@***.***> Cc: scelezy ***@***.***>, Comment ***@***.***> Subject: Re: [yihong0618/running_page] keep骑行数据导出为gpx文件 (Discussion #279) 
我之前也是和你一样的问题，在测试了RUN_DATA_API返回内容之后，突然就好了，你可以参考一下我keep_sync文件的代码：https://paste.ubuntu.com/p/b3D9sC2bFY/
